builder/simplify.cc

```diff
@@ -22,6 +22,18 @@ namespace slinky {
 
 namespace {
 
+// Returns true if e == select(e, 1, 0)
+bool is_boolean_valued(const expr& e) {
```

Comment:
Dumb question, but based on the comment, I'd expect this to also return true if the expr is literally a select node where both values are boolean-valued? (not a logic issue, just a comment issue)

Reply:
I started writing a visitor that implemented this transitively, but it gets a little tricky, and I decided to just keep it simple Even a moderately complicated visitor would not guarantee to successfully prove this in all cases anyways though. 

I don't think we have much code that will generate selects with nested/combined conditions, if any. All the code I'm looking at that I'm trying to simplify has deep nesting in the true/false value but never the condition.